assignments/prototype-refactor.js

```diff
@@ -7,3 +7,222 @@ Prototype Refactor
+      * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
+      * Instances of CharacterStats should have all of the same properties as GameObject.
+    */
+    
+    
+    
+    // Test you work by un-commenting these 3 objects and the list of console logs below:
+    
+    
+      const mage = new Humanoid({
+        createdAt: new Date(),
+        dimensions: {
+          length: 2,
+          width: 1,
+          height: 1,
+        },
+        healthPoints: 5,
+        name: 'Bruce',
+        team: 'Mage Guild',
+        weapons: [
+          'Staff of Shamalama',
+        ],
+        language: 'Common Tongue',
+      });
+    
+      const swordsman = new Humanoid({
+        createdAt: new Date(),
+        dimensions: {
+          length: 2,
+          width: 2,
+          height: 2,
+        },
+        healthPoints: 15,
+        name: 'Sir Mustachio',
+        team: 'The Round Table',
+        weapons: [
+          'Giant Sword',
+          'Shield',
+        ],
+        language: 'Common Tongue',
+      });
+    
+      const archer = new Humanoid({
+        createdAt: new Date(),
+        dimensions: {
+          length: 1,
+          width: 2,
+          height: 4,
+        },
+        healthPoints: 10,
+        name: 'Lilith',
+        team: 'Forest Kingdom',
+        weapons: [
+          'Bow',
+          'Dagger',
+        ],
+        language: 'Elvish',
+      });
+    
+      console.log(mage.createdAt); // Today's date
+      console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
+      console.log(swordsman.healthPoints); // 15
+      console.log(mage.name); // Bruce
+      console.log(swordsman.team); // The Round Table
+      console.log(mage.weapons); // Staff of Shamalama
+      console.log(archer.language); // Elvish
+      console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+      console.log(mage.takeDamage()); // Bruce took damage.
+      console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
+    
+    
+      // Stretch task: 
+      // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
+      // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
+      // * Create two new objects, one a villain and one a hero and fight it out with methods!
+    
+      //Hero
+
+      class Hero extends Humanoid {
+          constructor(atts) {
+            super(atts);
+            this.class = atts.class;
+            this.damage = atts.damage;
+            this.drops = atts.drops;
+          }
+          attack(who) {
+            if (who.name == this.name) {
+                console.log(`It'd be stupid to attack yourself ${this.name}`)
+                return
+              }
+              let minhit = 0;
```

Comment:
awesome job listing the variables here. 
At least in my opinion, this makes it so much easier to debug and reason about than the one liners.